docs/rules/max-statements-per-line.md

```diff
@@ -0,0 +1,152 @@
+# Limit Maximum Number of Statements Per Line (max-statements-per-line)
+
+A line of code containing too many statements can be difficult to read. Code is generally read from the top down, especially when scanning, so limiting the number of statements allowed on a single line can be very beneficial for readability and maintainability.
+
+```js
+function () { var bar; if (condition) { bar = 1; } else { bar = 2; } return true; } // too many statements
+```
+
+## Rule Details
+
+This rule allows you to configure the maximum number of statements allowed per line.  The default is 1.
+
+Examples of **incorrect** code for a maximum of 1:
+
+```js
+/*eslint max-statements-per-line: [2, 1]*/
+
+var bar = 1; var baz = 2;
+if (condition) var bar = 1; if (condition) var baz = 2;
+if (condition) { } if (condition) { }
+if (condition) { var bar = 1; } else { var bar = 1; }
```

Comment:
Three.

1. The `IfStatement` (with or without an alternate)
2. The `VariableDeclaration` inside the `BlockStatement` of the consequent
3. The `VariableDeclaration` inside the `BlockStatement` of the alternate

`BlockStatement`s that comprise the body of another statement (`IfStatement` consequent or alternate, `ForStatement`, etc.) do not count as their own statements (otherwise an `if / else` with both `BlockStatement`s would count as three alone). `BlockStatement`s that stand on their own (e.g. parent is a `BlockStatement` or `Program`) will count as a statement, however.

Do you agree with this behavior?